src/gc/impl/conservative/gc.d

```diff
@@ -252,6 +253,10 @@ debug (LOGGING)
 
 class ConservativeGC : GC
 {
+    bool profiling_enabled;
+    shared Mutex profiler_lock;
```

Comment:
Currenty main GC implementation uses non re-entrant spinlock. Anyway, all lock/sync hacking is more of proof of concept to see if I can make it work that way (have failed so far).